source/message.html

```diff
@@ -4,24 +4,82 @@
 <TITLE></TITLE>
 <SCRIPT LANGUAGE=javascript>
 <!--
-function escapeKeyPress(e) {
+// Though these should be constants, MSHTML does not support const as of 2024
+var args = window.dialogArguments;
+var copyStatusTimeout = 5000;  // milliseconds before copy status message goes away
+
+function handleKeyPress(e) {
 	e = e || window.event;
 	if(e.keyCode == 27){ // code for escape
 		window.close();
+	} else if (e.altKey && e.keyCode == 67) { // code for Alt+C
+		onCopyButtonPress();
+	}
+}
+
+function onCopyButtonPress() {
+	// Copy code came from http://www.codestore.net/store.nsf/unid/DOMM-4QHQE8/
+	var rng = document.body.createTextRange();
+	rng.moveToElementText(messageID);
+	rng.scrollIntoView();
+	rng.select();
+	var success = rng.execCommand("Copy");
+	rng.collapse(false);
+	rng.select();
+	if (args) {
+		if (success) {  // Notify the user about the copy result
+			copyStatusDiv.innerHTML = '<p id="copyStatusText"><strong>' + args.item('copySuccessfulAlertText') + '</strong></p>';
+		} else {
+			copyStatusDiv.innerHTML = '<p id="copyStatusText"><strong>' + args.item('copyFailedAlertText') + '</strong></p>';
```

Comment:
can these args be escaped somehow before adding to html?

Reply:
I will work around their unescapedness by adding them as text, which is hopefully sufficient?

Reply:
No, you'd need to escape any HTML characters if adding the text using `innerHTML`.
You may need to find a legacy solution for MSHTML JS
https://stackoverflow.com/questions/6234773/can-i-escape-html-special-chars-in-javascript

It might be easier to do it from python with the module `nh3`, if you need to use `innerHTML` for adding translated strings

Reply:
@seanbudd I have been able to switch to `.innerText` for everything but the case of the consumer passing in a message it intends to be HTML.
For that case, at this stage, I'm logging a warning on the NVDA side, and otherwise passing it unchanged.

I don't have any objection to using NH3, but I think:
* That would need some implementation discussion as it's got some configurability;
* I would consider it API breaking, since we can't say what behavior add-ons are depending on in raw HTML that they have been passing, that we would now sanitize out; and
* I would prefer it was its own PR.

Particularly in relation to my last point: I think this one leaves us no worse off than we were before, if not better since this has better error handling if nothing else.

Reply:
Thanks, I agree that it is probably out of scope of this PR